### PR TITLE
Added docstring for ExprWithIntLimits class

### PR DESCRIPTION
**Pull request by czgdp1807**
#### References to other Issues or PRs

Closes #11931

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/concrete/expr_with_intlimits.py [hunk 1]
```diff
@@ -13,6 +13,14 @@ def __init__(self, expr, msg):
             "%s could not be reordered: %s." % (expr, msg))
 
 class ExprWithIntLimits(ExprWithLimits):
+    """
+    The class for expressions with integer limits.
```
Comment (asmeurer) [inline sympy/concrete/expr_with_intlimits.py]: This should specify that it is the superclass of Sum and Product.

Comment (asmeurer) [reply sympy/concrete/expr_with_intlimits.py]: See https://github.com/sympy/sympy/pull/17698/files#diff-50952a7e18ce9c5bd3f6b4b0299b8174

Comment (czgdp1807) [reply sympy/concrete/expr_with_intlimits.py]: I have made an update. Please take a look.

**Comment by czgdp1807** [comment]
@asmeurer Can we merge this?

**Comment by asmeurer** [comment]
Please wait until my PR is merged first.

**Pull request merged**